autogen/README.md

```diff
@@ -0,0 +1,86 @@
+# Autogen Framework Model Interfaces
+The Autogen framework gives 3 major file based interfaces for models: `model.json`, `modelparameters.m`, and `sm_stop_verify.m` and 1 major interface available as a struct in the Matlab workspace: `mp`.
+
+## model.json
+`model.json` is a required configuration file shared between the Autogen Framework and the Autogen Tools. This covers both system-level properties as well as run-time tunable parameters for the model via registers.  
+`model.json` properties:  
+- `system`: An object to describe the overall system  
+    - `target`: Which hardware system is being targeted, the `audiomini` or `audioblade` (default: `audiomini`)  
+    - `sampleClockFrequency`: The sampling rate of the audio being processed (default: `48000`)  
+    - `systemClockFrequency`: Clock rate of the audio processing on the FPGA (default: `98304000`)  
+    - `inputDataType`: Object representing the data type of the audio input  
+        - `wordLength`: number of total bits in the data type  
+        - `fractionLength`: number of fractional bits in the fixed point number  
+        - `signed`: true if the data type is signed, false otherwise  
+    - `outputDataType`: Object reprenting the data type of the audio output
+    (Note: Currently unused, only inputDataType is read)
+        - `wordLength`: number of total bits in the data type  
+        - `fractionLength`: number of fractional bits in the fixed point number  
+        - `signed`: true if the data type is signed, false otherwise  
+    - `numberOfChannels`: number of channels in the audio to be processed (default: `2`)  
```

Comment:
`numberOfChannels` should be moved into the `inputDataType` and `outputDataType` properties. The [beamforming model configuration file](https://github.com/fpga-open-speech-tools/simulink_models/blob/dev/models/delay_and_sum_beamformer/DSBF_dataplane.json) has different numbers of channels on input and output

Reply:
Sounds good. Additionally, is that part of a data type really or would renaming `inputDataType` and `outputDataType` to something like `audioIn` and `audioOut` make more sense?

Reply:
I think the renaming makes more sense. The number of channels isn't really part of the data type. It could be 
```json
audioIn:
  {
    dataType: {
    }
    numberOfChannels: 2
}
```
or we could flatten it.

Reply:
Implemented the flattened version of this